### PR TITLE
zigbee: add ncp custom command for changing led states

### PR DESCRIPTION
**Pull request by FilipZajdel**
Exemplary usage of ncp custom commands to control the state
of a led and sending indications containing the led index.

Signed-off-by: Filip Zajdel <filip.zajdel@nordicsemi.no>

### CODE REVIEW

File: samples/zigbee/ncp/src/main.c [hunk 1]
```diff
@@ -68,6 +92,92 @@ void zb_osif_bootloader_report_successful_loading(void)
...
+}
+
+static zb_ret_t ncp_vendor_specific_req_handler(zb_uint8_t buf)
+{
+	/* request tsn */
+	zb_uint8_t tsn = *ZB_BUF_GET_PARAM(buf, zb_uint8_t);
+	/* actual payload passed by the request */
+	zb_uint8_t *led_state = (zb_uint8_t *)zb_buf_begin(buf);
+
+	zb_uint8_t resp_buf = zb_buf_get(ZB_FALSE, VENDOR_SPECIFIC_RESPONSE_LEN);
```
Comment (sebastiandraus) [inline samples/zigbee/ncp/src/main.c:112]: Check return value of `zb_buf_get`

**Review by sebastiandraus — CHANGES_REQUESTED** [review]

File: samples/zigbee/ncp/src/main.c [hunk 2]
```diff
@@ -68,6 +88,89 @@ void zb_osif_bootloader_report_successful_loading(void)
...
+
+	zb_ncp_custom_indication(buf);
+}
+
+static void perform_custom_indication(zb_uint8_t arg)
+{
+	ZVUNUSED(arg);
+
+	zb_buf_get_out_delayed_ext(custom_indication, vendor_specific_led_state, 0);
+}
```
Comment (sebastiandraus) [inline samples/zigbee/ncp/src/main.c:105]: Please fix this:

```suggestion
static void perform_custom_indication(zb_uint8_t led_idx)
{
	zb_buf_get_out_delayed_ext(custom_indication, led_idx, 0);
}
```

**Review by sebastiandraus — CHANGES_REQUESTED** [review]

File: samples/zigbee/ncp/src/main.c (hunk 2 above)
Comment (FilipZajdel) [reply samples/zigbee/ncp/src/main.c]: fixed

**Review by sebastiandraus — APPROVED** [review]

**Review by maciejpietras — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Comment by SebastianBoe** [comment]
Awaiting CI

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
No need to wait @SebastianBoe, this passed CI in an earlier run.

**Pull request merged**